js/foundation.abide.js

```diff
@@ -59,6 +61,14 @@ class Abide extends Plugin {
         return this.validateForm();
       });
 
+    this.$submitDisabler
+      .off('click.zf.abide')
+      .on('click.zf.abide', (e) => {
+        e.preventDefault();
+        this.enableValidation(false);
+        this.$element.submit();
+      });
```

Comment:
The `[formnovalidate]` attribute on an input element make the validation being bypassed _when the form is submitted via this element_. This does not enable or disable the form validation entirely and does not affect others validation methods.

See https://codepen.io/ncoden/pen/LXJybV?editors=1111

So if we want to support this attribute for Abide, we should search for a `[formnovalidate]` attribute on the `currentTarget` during the submit event, and bypass the validation if found. 

Also, any input can have `[formnovalidate]`, since using <kb>Enter</kb> on an `input[type="text"]` will trigger the form submission.

This does not prevent to have a private `isValidationEnabled` state next to that, or to dynamically check at each submission for `data-enable-validation` on the form element. But it should not be affected by `[formnovalidate]`, which is specific to the trigger of the form submission.

Reply:
> This does not enable or disable the form validation entirely

It does! When the submit is triggered by an input with `formnovalidate` the whole validation is skipped (`required`, `type="email"`etc.)
Or have I misunderstood you?

> we should search for a `[formnovalidate]` attribute on the `currentTarget` during the submit event

The `currentTarget` would be the form which doesn't have any `formnovalidate` attribute. I don't see another way than listening to the submit input/button.

> Also, any input can have `[formnovalidate]`, since using Enter on an `input[type="text"]` will trigger the form submission.

It's useless on any other input than the submit input. You can test is in your codepen:
- remove `formnovalidate` from the submit input
- add `formnovalidate` to the required email input

The validation won't get bypassed.
Pressing the Enter key will trigger the next submit input in the form's markup.
Imagine there are two submit inputs in the form, after the required email input.
- if the first submit input has `formnovalidate` the validation is bypassed
- if only the second submit input has `formnovalidate`(and the first not) the validation is not bypassed

Reply:
>  Or have I misunderstood you?

I afraid so. I meant that if you have a form with 2 submit inputs, one with `[formnovalidate]` and one without, submitting the form once with the first input will only disable the validation for this submission, and will not disable it for a second submission later via the second input. It's why I talked about a "form-level state" vs a "submission-level state". See the provided Codepen.

> The `currentTarget` would be the form

You are right, I forgot about that. I wanted to retrieve the element that triggered the form submission to handle `[formnovalidate]` like the browser does. In Firefox this would be `originalEvent.explicitOriginalTarget` (but it's an experimental feature).

> It's useless on any other input than the submit input

Again you are right. Only `input[type="submit"]` elements are taken into account for `[formnovalidate]`. When entering <kb>Enter</kb> on `input[type="text"]`, the event will be intercepted by the first `input[type="submit"]` element found in the form and the `submit` event will be submitted from it, then retargeted by the form element (so we loose the trace of the original submit input).

The hyperform package use the [following helper](https://github.com/hyperform/hyperform/blob/master/src/tools/catch_submit.js) to retrieve the submit button that were used to submit the form. It's a bit heavy though.

---

My point is only that whether the validation will be bypassed on depends on the submit button used directly or indirectly for the current submission only, regardless of "whether the form validation is enabled" (at the form level).

Reply:
> submitting the form once with the first input will only disable the validation for this submission, and will not disable it for a second submission later via the second input.

if the form is submitted by the first input (bypassed validation) the page will probably reload so it doesn't matter if the validation is enabled again afterwards or not.
But I got your point – the disabled validation should be bound to the `formnovalidate` input and not affect the other inputs.

> I wanted to retrieve the element that triggered the form submission

Since all those elements can be found by `[type="submit"][formnovalidate]` I think the safest way is listening to those elements.

> The hyperform package use the following helper to retrieve the submit button that were used to submit the form. It's a bit heavy though.

Too heavy – I prefer event listener on submit inputs as described above.

> My point is only that whether the validation will be bypassed on depends on the submit button used directly or indirectly for the current submission only, regardless of "whether the form validation is enabled" (at the form level).

Ok, got it! I'll try to emulate this behavior.

Reply:
> Since all those elements can be found by [type="submit"][formnovalidate] I think the safest way is listening to those elements.

To behave like the native `[formnovalidate]` have to support:
- Click on the submit input
- <kb>Space</kb>/<kb>Enter</kb>
- <kb>Enter</kb> on any field of the form
- Any other form submission process

This is heavy. I would recommend you to drop the `[formnovalidate]` support and simply use a dynamic option (checked at the form submission) like `data-validate`. What do you think?

Reply:
I've reworked this PR: Abide behaves now 100% identical to native formnovalidate